contents/security/crossSiteRequestForgery/crossSiteRequestForgery.md

```diff
@@ -0,0 +1,96 @@
+<input type="number" name="amount" value="100"/>
+<input type="submit"/>
+</form>
+```
+
+As Alice clicks the submit button, the browser will sending the following request to `www.example-bank.com`:
+
+```
+POST http://www.example-bank.com/transfer HTTP/1.1
+Cookie: auth=1abcd2ek3292fsa390sdf
+
+account-no=123456&amount=100
+```
+
+The bank's server verifies that the cookie `auth=1abcd2ek3292fsa390sdf` is valid and is associated with Alice. It then proceeds to transfer $100 from Alice's account to account number 123456, which belongs to Bob.
+
+Alice then receives a chat message from her not-so-trustworthy friend, Eve:
+
+```
+Eve: OMG Alice! This site is selling the shoes you have always wanted for half the price www.i-am-not-evil.com/shoes
+```
+
+Excited by this piece of news, Alice clicks on the hyperlink to checkout `www.i-am-not-evil.com`. She then clicks on the `View more pictures` button on the website to see more pictures of her favorite pair of shoes. Unbeknownst to her, that button has the following HTML code:
+
+```
+<form action="http://www.example-bank.com/transfer" method="POST">
+<input type="hidden" name="account-no" value="987654"/>
+<input type="hidden" name="amount" value="100000"/>
+<input type="submit" value="View more pictures"/>
+</form>
+```
+
+So, when she clicks that button, the browser actually sends the following request to `http://www.example-bank.com`:
+
+```
+POST http://www.example-bank.com/transfer HTTP/1.1
+Cookie: auth=1abcd2ek3292fsa390sdf
+
+account-no=987654&amount=100000
+```
+
+Since Alice is still logged in to `www.example-bank.com`, any request made to `www.example-bank.com` is automatically attached the cookie `auth=1abcd2ek3292fsa390sdf`. Hence, the bank's server considers this as an authenticated request from Alice and proceeds to transfer $100,000 from Alice's account to account number 987654, which belongs to the attacker! Indeed, `www.i-am-not-evil.com` could have performed the attack by using Javascript to submit the form, which would eliminate the need for Alice to click on anything, so she will not even be aware of it.
+
+## Conditions for a successful CSRF attack
+
+From the example, we can see that a CSRF attack works by forging a valid request which **inherits the identity and privileges of the victim** to the vulnerable web server. Such inheritance of privileges is possible because the browser attaches any credentials associated with a website to all requests made to the site. Therefore, if the user is currently authenticated to the site, the site will have no way to distinguish between the forged request sent by the victim and a legitimate request sent by the victim. In short, there are 2 conditions necessary for the attack:
+
+1. The target website is vulnerable to a CSRF attack (i.e. it cannot distinguish between forged and legitimate requests).
+2. The victim is authenticated to the target website.
+3. The victim is tricked to visit an untrusted website by social engineering.
```

Comment:
Same problem as the `cookie` issue here - the term `social engineering` is wholly unexplained and would benefit from either having a hyperlink to some definition or being introduced in a previous section